pandas/core/indexes/base.py

```diff
@@ -1385,6 +1408,16 @@ def set_names(self, names, level=None, inplace: bool = False):
             idx = self
         else:
             idx = self._shallow_copy()
+
+        if isinstance(self, ABCMultiIndex) and is_dict_like(names):
+            # Transform dict to list of new names and corresponding levels
+            level, names_adjusted = [], []
+            for i, name in enumerate(self.names):
```

Comment:
see my comment above i think this condition needs to be on L1402

Reply:
Moved it up a bit and added level is None